tests/components/stt/test_init.py

```diff
@@ -444,3 +446,77 @@ async def test_ws_list_engines(
     assert msg["result"] == {
         "providers": [{"engine_id": engine_id, "supported_languages": ["de-CH", "de"]}]
     }
+
+
+async def test_default_provider_none(hass: HomeAssistant, tmp_path: Path) -> None:
```

Comment:
Maybe move this to the legacy test module, `test_legacy.py`? I've tried to put tests there that only are useful for the legacy code, so that we can easily remove these when removing the legacy code.

Reply:
I see you changed the code so my comment isn't relevant anymore.